### ISSUE TITLE
Observations from a reconciliation process

### ISSUE THREAD

**Issue by susannaanas** [OpenRefine/OpenRefine#2083]
**Describe the bug**
I created a memo of problems with a reconciliation process. I will break it down to proper issues after finishing the project and possible comments, corrections and removals.

**Desktop (please complete the following information):**
 - OS: MacOS 10.13.6
 - Browser Version: Version 75.0.3770.100 (Official Build) (64-bit)
 - JRE or JDK Version:[output of "java -version" e.g. JRE 1.8.0_181]

**OpenRefine (please complete the following information):**
 - Version 3.2-SNAPSHOT [8d89a2a]

**Datasets**
https://drive.google.com/open?id=1_i-Gy2Q584I9V0ktiGdp2ME9qt5vvedk

**Additional context**
When a mass import needed to be stopped to make a correction, the OpenRefine database and Wikidata got out of sync. The already added Wikidata information was not updated to the OpenRefine items that had already been uploaded. On another occasion, they did not get updated although the update ran normally. To my knowledge, the new QID should be matched to the OR item, but if that is not the case, updating reconciliation data would be an enhancement request. The new items are however created in Wikidata and this will result in a new matching process.

It should be possible to match more explicitly based on an external identifier. It is ok for some cases that there is a fallback mechanism for a regular match, but that should be possible to switch off/control. It is difficult to know which of the matches are certain, and which are good guesses. This could also be solved by indicating the type of match in the UI.

The popup matching interface is somewhat limited. If the item is not in the first set of suggestions in the popup, there is no way to retrieve more.

There could be visual interfaces for geolocations. Maybe even the distance between the compared geocoordinates could be retrieved rather than the mysterious score.

Selecting an option in the popup makes the page jump to the first page of a series of possibly hundreds of pages. This remarkably slows down work. I know this is a known bug and I will give my thumb to correcting it.

The new matching interface that forces to use the popup is very annoying, one reason above. The checkboxes do not refresh the page in a similar manner.

Key columns with values in multiple rows causes misery and sorrow. It is unclear how records with multiple rows should be handled.
1 The key column needs to be filled down to be able to add the multiple properties to the item (I think). If any of the values for properties is declared within the schema rather than in the data, it is added to all the split rows, multiple times to one item.
2 When using the “Choose new match” popup for such a filled-down item with multiple rows, it is unclear if a new value is created once to the whole record or multiple times for each row. This is made vaguely distinguishable in the popup UI, but it is not displayed in any way after adding to be able to track errors.

It is unclear how to manage new items as values for properties in the schema. How to give the new valueitems necessary properties to be able to use them as values in the schema. It seems it should be made an additional schemaitem in the schema, but in which order is data read and created is not clear. This has resulted in errors, but I need to dig into archives to find the specifics. 

→ It is unclear how to manage multiple items in one import schema.

The file gets slow over time. The popups are impossible to use as they do not populate. it becomes impossible to reconcile. A workaround could be that much of the information of the reconciliation options would be imported to the file that are now only shown in the popups dynamically.

Local copy: https://docs.google.com/document/d/124QsXwcyv3hEGheT6WygdpNZADVRQCYOoZgKroHyw2Q/edit?usp=sharing

**Comment by wetneb** [comment]
Many thanks for this very detailed feedback, it is very useful! I will try to reply with more detailed answers and links to existing issues later.

**Comment by wetneb** [comment]
**Updating OR in the reconciliation process**

Yes, reconciliation results can get out of date and this can be an issue. I do not really see what we could do against this problem: the current solution is to re-run reconciliation with the same configuration, how could this be improved?

Concerning errors during editing, we have #1782 where we need to figure out the right design.

To create a numeric facet on the Qids, you can create a "Custom numeric facet" and use an expression such as `cell.recon.match.id.substring(1).toNumber()`.

**Matching by external IDs**

These suggestions are extremely interesting and I will post them on the mailing list of the [W3C Entity Reconciliation Community Group](https://www.w3.org/community/reconciliation/) (that you are welcome to join if you like!)

**Popups**

As you know we have #33 already but all the other problems you mention are worth having issues on their own I would say.

**Multiple rows in a record**

Again the problems you mention would definitely be worth having individual issues.

**Miscellaneous**

> There could be visual interfaces for geolocations.

I guess this would mean adding some sort of maps in the item previews? If so then that belongs to https://github.com/wetneb/openrefine-wikibase.

> Maybe even the distance between the compared geocoordinates could be retrieved rather than the mysterious score.

That is typically something I want to push for in the W3C group, following my recommendations in https://arxiv.org/pdf/1906.08092

**Comment by tfmorris** [comment]
This is good feedback, but needs to be split into multiple actionable issues for things to be improved. I'm not sure how many. Some of the complaints I recognize as duplicating existing issues, e.g. the 8 year old #33, but someone needs to go through them all one by one.

**Comment by thadguidry** [comment]
@tfmorris Agree, so perhaps label this as "to be reviewed" for whenever someone has that time later.

**Comment by tfmorris** [comment]
Label added, but feel free to do that yourself in the future.

**Comment by pyrog** [comment]
> **Matching by external IDs**
> It should be possible to match more explicitly based on an external identifier.
👍 

Also, **matching against properties** seem useful too.

In [World Port Index](https://msi.nga.mil/Publications/WPI), the port have a ISO country code.
Matching it against [country (Q6256)](https://www.wikidata.org/wiki/Q6256) don't work, but matching it against  ISO [3166-1 alpha-2 code (P297)](https://www.wikidata.org/wiki/Property:P297) will find all countries.

An new option "Reconcile against property" could show the column in the right pane, so we could "include it as property" 
[P297](https://www.wikidata.org/wiki/Property:P297).

**Comment by pyrog** [comment]
> **Miscellaneous**
> There could be visual interfaces for geolocations.

👍 Could display an OpenStreetMap map, but at which zoom level ?

**Comment by pyrog** [comment]
> **Miscellaneous**
>  Maybe even the distance between the compared geocoordinates could be retrieved rather than the mysterious score.

👍 Could compute a simple [Geographical distance](https://en.wikipedia.org/wiki/Geographical_distance) with a distance parameter setting
